### PR TITLE
chore: Remove Docs from OSx Repo

### PR DESCRIPTION
**Pull request by banasa44**
## Description

This PR has the complete removal of OSx hand-written documentation from OSx. With its necessary changes in the workflows.

It also adds a checkbox in the PR template so from now on, anyone who makes important changes in the OSx repo should create a follow-up task to reflect those changes in the Developer Portal.

At the end, there are some orthographic corrections in the Lifecycle .md files.

Please take a close look into the GitHub files, and see at the end if the orthographic corrections are correct.

Task ID: [OS-1352](https://aragonassociation.atlassian.net/browse/OS-1352)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.
- [ ] I have updated the Subgraph and added a QA URL to the description of this PR.

[OS-1352]: https://aragonassociation.atlassian.net/browse/OS-1352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/subgraph-documentation-update.yml [hunk 1]
```diff
@@ -42,10 +42,10 @@ jobs:
           path: developer-portal
           token: ${{ secrets.ARABOT_PAT }}
       - name: Remove the docs/subgraph folder in the developer-portal
-        run: rm -rf $GITHUB_WORKSPACE/developer-portal/docs/subgraph
+        run: rm -rf $GITHUB_WORKSPACE/developer-portal/versioned_docs/version-1.3.0/osx/subgraph
       - name: Copy the docs/developer-portal folder to the developer-portal
-        run: cp -R packages/subgraph/docs/developer-portal $GITHUB_WORKSPACE/developer-portal/docs/subgraph
-      - name: Copy the generated schema-introspection.json to the developer-portal
+        run: cp -R packages/subgraph/docs/developer-portal $GITHUB_WORKSPACE/developer-portal/versioned_docs/version-1.3.0/osx/subgraph
```
Comment (Michael-A-Heuer) [inline .github/workflows/subgraph-documentation-update.yml:47]: Shouldn't this be 1.4.0?

**Review by Michael-A-Heuer — CHANGES_REQUESTED** [review]
Make sure the version is correct

File: .github/workflows/subgraph-documentation-update.yml (hunk 1 above)
Comment (banasa44) [reply .github/workflows/subgraph-documentation-update.yml:47]: For the moment we agreed to have both SDK and Subgraph living inside OSx's 1.3.0. 
Moreover with the refactoring 1.4.0 will be the new discussed format and we didn't talk about including Subgraph.

File: packages/contracts/docs/developer-portal/04-framework-lifecycle/02-semver.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ See the [OpenZeppelin article on multiple inheritance](https://docs.openzeppelin
 - Change
 
   - Body
-    A change to a function body can change the behaviour of the function requiring dependent contracts or software systems to change
```
Comment (clauBv23) [inline packages/contracts/docs/developer-portal/04-framework-lifecycle/02-semver.md:111]: Line 89 `overridden` also has a typo
``` 
 - methods affected must be virtual to be overridden
 ```

**Review by clauBv23 — APPROVED** [review]

**Review by clauBv23 — APPROVED** [review]

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**